File: app/src/main/java/com/example/myownframework/KOTLIN_ANDROID/CollectionFrameWork/java/Enumeration.java
package com.example.myownframework.KOTLIN_ANDROID.CollectionFrameWork.java;


/*1. Enumeration (Interface)
Enumeration is an interface in Java, which was part of the original JDK 1.0.
 It is used to iterate over elements in collections like Vector, Hashtable, and other legacy classes.*/


import java.util.Hashtable;
import java.util.Vector;

public class Enumeration {

    public static void main(String[] args) {
        Hashtable<String,Integer> hashtable = new Hashtable<>();
        hashtable.put("srinivas",1);
        hashtable.put("kumar",2);
        hashtable.put("reddy",3);

        java.util.Enumeration<Integer> enumeration =  hashtable.elements();
        while(enumeration.hasMoreElements()){
            System.out.println(enumeration.nextElement());
        }


        // second
        Vector<String> vector = new Vector<>();
        vector.add("srinivas");
        java.util.Enumeration<String> enumeration1 =  vector.elements();
        while(enumeration1.hasMoreElements()){
            System.out.println(enumeration1.nextElement());
        }
    }


}

File: app/src/main/java/com/example/myownframework/KOTLIN_ANDROID/javajava/collectionframework/Cursor/Enumeration.kt
package com.example.myownframework.Kotlin_Android.javajava.collectionframework.Cursor

import java.util.Vector

fun main(){

    val vector : Vector<Int> = Vector()
    for (i in 0..10){
        vector.addElement(i)
    }
    println(vector)

    val e = vector.elements()
    println(e.javaClass)
    while (e.hasMoreElements()) {
        val i = e.nextElement()
        println(i)
    }
}